### DISCUSSION TITLE
Z-Wave JS configurazione

### DISCUSSION THREAD

**Discussion by baldus88** [indomus/forum#1637] [ZigBee, Z-Wave, Thread, Matter]
Ciao ragazzi.
Ripropongo un po' quello che avevo anticipato su facebook dopo il consiglilo dei moderatori di dirigermi qui.

Sto migrando, con un po' di ritardo, la configurazione zwave (fortunatamente ad ora non ho componenti attivi quindi meno impatto) dalla vecchia config a zwave js.
Ho istallato il docker e docker compose e ho proseguito a seguire la guida:
[https://indomus.it/.../installare-e-configurare-z.../...](https://indomus.it/guide/installare-e-configurare-z-wave-js-con-docker-su-raspberry-pi-os/?fbclid=IwAR2JczWK9O9Areuve-4dgy5YE9RHUJCd9TkAthYQnFjAmx69RxM7_bI3ttk#port)

Detto che inizialmente avevo installato il container senza composer e avevo dei dubbi sul fatto che la porta fosse giusta.

Ho comunque provato a lanciare lo script per cercare la mia scheda razberry board ma escono solo le porte usb:
```
pi@raspberrypi:~ $ ./ports.sh
/dev/bus/usb/001/001 - Linux_5.10.103-v7+_dwc_otg_hcd_DWC_OTG_Controller_3f980000.usb
/dev/bus/usb/001/003 - 0424_2514
/dev/bus/usb/001/005 - 0424_7800
/dev/bus/usb/001/002 - 0424_2514
/dev/ttyUSB0 - 1a86_USB_Serial
/dev/bus/usb/001/004 - 1a86_USB_Serial
```
Dalla configuazione precendente io usavo questa: /dev/ttyAMA0, ma qua non la trovo.

Infatti provando a lanciare il comando avevo questo errore:
`
docker run --init -d -it --name zwavejs2mqtt -p 8091:8091 -p 3000:3000 --PORTA_COORDINATOR:/dev/zwave -v $/home/pi/zwavejs2mqtt:/usr/src/app/store --restart always zwavejs/zwavejs2mqtt:latest
`
Inserendo la porta mi da questo errore:
`
unknown flag: --/dev/ttyAMA0
`

Successivamente ho usato la /dev/ttyUSB0 e pare che sia partito.

```
pi@raspberrypi:~/compose $ docker logs zwavejs2mqtt
/bin/sh: git: not found
2022-04-11 21:03:20.608 INFO APP: Version: 6.7.0
2022-04-11 21:03:20.618 INFO APP: Application path:/usr/src/app
2022-04-11 21:03:20.683 WARN STORE: settings.json not found
2022-04-11 21:03:20.688 WARN STORE: scenes.json not found
2022-04-11 21:03:20.691 WARN STORE: nodes.json not found
2022-04-11 21:03:20.722 WARN STORE: users.json not found
2022-04-11 21:03:20.767 INFO APP: Listening on port 8091 host 0.0.0.0 protocol HTTP
2022-04-11 21:03:21.333 WARN ZWAVE: Zwave driver not inited, no port configured
```

Ma continua a darmi "Zwave driver not inited, no port configured" è normale?

Provando a navigare la configurazione web e fare gli step della guida ho questo errore di inizializzazione:
![image](https://user-images.githubusercontent.com/57867040/163050156-5ed06c78-b71f-4059-ad3b-6282c33bdf84.png)

Potete darmi qualche supporto?

Grazie!

**Comment by KC-inDomus** [comment]
`unknown flag: --/dev/ttyAMA0`

stai sbagliando il comando, che invece dev'essere:

`docker run --init -d -it --name zwavejs2mqtt -p 8091:8091 -p 3000:3000 --device=/dev/ttyUSB0:/dev/zwave -v`

come da guida.

**Comment by baldus88** [comment]
Ciao.

Ho eliminato il container e l'ho rigenerato così:
```
version: "3.9"
services:
   zwavejs2mqtt:
    container_name: zwavejs2mqtt
    image: zwavejs/zwavejs2mqtt:latest
    restart: unless-stopped
    devices:
      - "/dev/ttyUSB0:/dev/zwave"
    volumes:
      - "/home/pi/zwavejs2mqtt:/usr/src/app/store"
    ports:
      - "3000:3000"
      - "8091:8091"
```
In aggiunta questo path: /usr/src/app/store ho provato a cercare in giro ma lo devo creare così com'è? Perchè di default chiaramente non ce l'ho e la procedura non lo crea. Con quali permessi nel caso?

Nel caso comunque sia l'errore rimane lo stesso

**Reply by KC-inDomus** [reply]
a parte la porta (che evidentemente è sbagliata), la config è giusta. non ti sta vedendo l'antenna, è quello il problema imho. ma il vecchio software lo hai femato, sì?

**Reply by baldus88** [reply]
Quando configurai l'integrazione zwave che io ricordi non avevo fatto niente di particolare a parte disabilitare il bluetooth modem service ed abilitare la serial interface con questo comando:
sudo systemctl mask serial-getty@ttyAMA0.service

Il network management della rete z-wave non lo vedo più dalla dashboard avendo aggiornato home assistant una versione troppo recente e quella integrazione non la vedo più.

Posso verificare nel caso in qualche altro modo?
Per verificare invece la porta corretta a parte lo script citato come posso fare?

Grazie

**Comment by baldus88** [comment]
Ciao.
Ho ho provato a rimuovere il modulino razberry per vedere cosa mi estraesse lo script ma è rimasto tutto invariato:
`/dev/bus/usb/001/001 - Linux_5.10.103-v7+_dwc_otg_hcd_DWC_OTG_Controller_3f980000.usb
/dev/bus/usb/001/003 - 0424_2514
/dev/bus/usb/001/005 - 0424_7800
/dev/bus/usb/001/002 - 0424_2514
/dev/ttyUSB0 - 1a86_USB_Serial -> questa è la CC2538
/dev/bus/usb/001/004 - 1a86_USB_Serial  -> stessa di sopra`

Quindi probabilmente è come dicevo inizialmente che lo script estrae solo le porte usb collegate e non il modulo direttamente sulla scheda madre. Come faccio a capire che periferica ha? Quando usavo la configurazione zwave andava con questa /dev/ttyAMA0...

C'è qualcosa che sto sbagliando comunque... ho preso spunto dall'esempio sul sito e ho adattato la config:

version: "3.9"
services:
   zwavejs2mqtt:
    container_name: zwavejs2mqtt
    image: zwavejs/zwavejs2mqtt:latest
    restart: unless-stopped
    devices:
      - "/dev/ttyAMA0:/dev/zwave"
    volumes:
      - "/home/pi/zwavejs2mqtt:/usr/src/app/store"
    ports:
      - "3000:3000"
      - "8091:8091"

Sono arrivato fino al deploy corretto:
`Status: Downloaded newer image for zwavejs/zwavejs2mqtt:latest
Creating zwavejs2mqtt ... done
`
Solo che se faccio un listato dei log:
`pi@raspberrypi:~/compose $ docker container logs zwavejs2mqtt -f`
/bin/sh: git: not found
2022-05-05 21:20:20.112 INFO APP: Version: 6.9.0
2022-05-05 21:20:20.127 INFO APP: Application path:/usr/src/app
2022-05-05 21:20:20.245 WARN STORE: scenes.json not found
Logging to file:
        /usr/src/app/store/zwavejs_2022-05-05.log
Logging to file:
        /usr/src/app/store/zwavejs_2022-05-05.log
Logging to file:
        /usr/src/app/store/zwavejs_2022-05-05.log
Logging to file:
        /usr/src/app/store/zwavejs_2022-05-05.log
Logging to file:
        /usr/src/app/store/zwavejs_2022-05-05.log
Logging to file:
        /usr/src/app/store/zwavejs_2022-05-05.log

Mi esce questo path che tra l'altro sotto /usr/src/ nemmeno esisteva /app/store. Ma è giusto quello definito qui:
https://indomus.it/guide/installare-e-configurare-z-wave-js-con-docker-su-raspberry-pi-os/

O meglio che lo riadatto? Ho provato pure a crearlo ma i log non li crea anche cambiando permessi alla cartella:
pi@raspberrypi:~/compose $ cat /usr/src/app/store/zwavejs_2022-05-05.log
cat: /usr/src/app/store/zwavejs_2022-05-05.log: No such file or directory

Nel mentre se provo l'interfaccia web sta su ma continua a dare la porta vecchia addirittura:
![image](https://user-images.githubusercontent.com/57867040/167028094-c12b94d1-9dca-401a-9e19-6d9433f2b4ea.png)



Potete darmi una mano? 
Grazie

**Reply by KC-inDomus** [reply]
comincia col capire quale sia la porta che viene mappata quando il modulto è installato. concentrati solo su questo. è inutile che vai avanti a tentoni se ancora non sai neppure quale sia la porta.

**Reply by baldus88** [reply]
Ciao.
Non ho trovato il comando adatto per identificare il modulo, puoi darmi una mano?
Come avevo anticipato nel post avevo seguito la guida d'istallazione con determinati step (disablitare il bluetooth, configurare il servizio etc..)
https://wiki.fhem.de/wiki/Raspberry_Pi_3:_GPIO-Port_Module_und_Bluetooth

Dove mi aveva fatto configurare la /dev/ttyAMA0 che infatti nelle versioni precedenti di home assistant con l'integrazione oramai obsoleta zwave vedeva.
Chiaramente se faccio così c'è ancora:
pi@raspberrypi:~ $ ls -lha /dev/ttyAMA0
crw-rw---- 1 root dialout 204, 64 May 12 13:17 /dev/ttyAMA0

Ma inserendola nel file di configurazione di zwavejs2mqtt mi da lo stesso problema:
![image](https://user-images.githubusercontent.com/57867040/170831844-c6769e7a-a407-4ce3-aa2e-b3cf7c61d327.png)

E' più chiaro il contesto?

Grazie

**Comment by baldus88** [comment]
Ciao.
Giusto per fare la chiusa alla fine sono riuscito configurando la porta come avevo già scritto in precedenza (/dev/ttyAMA0):
```
version: "3.9"
services:
  zwavejs2mqtt:
    container_name: zwavejs2mqtt
    image: zwavejs/zwavejs2mqtt:latest
    restart: always
    tty: true
    stop_signal: SIGINT
    networks:
      - zwave
    devices:
      - '/dev/ttyAMA0:/dev/ttyAMA0'
    volumes:
      - ./store:/usr/src/app/store
    ports:
      - "8091:8091"
      - "3000:3000"
networks:
  zwave:
```

Dopo un paio di riavvi e alcuni permessi alle cartelle è andato e sono riuscito a migrare correttamente l'integrazione da zwave a zwavejs2mqtt dalla console di home assistant con la release 2022.3.6.

Ora è tutto regolare:
![image](https://user-images.githubusercontent.com/57867040/173695159-31f9a3a9-12c6-448f-ace3-04ee8f3b8c10.png)

Questo lo stato del webserver zwavejs:
![image](https://user-images.githubusercontent.com/57867040/173695189-3c9e63cb-f205-4572-86e5-e9d5151d2296.png)

Magari la mia esperienza può servire a qualcuno che ha la mia stessa componente o affine.

Grazie ancora del supporto.
